### ISSUE TITLE
Reader: Delay when tapping back from fullpost view

### ISSUE THREAD

**Issue by rachelmcr** [wordpress-mobile/WordPress-iOS#5981]
### Expected behavior

I expect that when I tap from a fullpost view back to the post list view in the Reader, the back action will take effect immediately.

### Actual behavior

There can be a slight delay between tapping the back button and being taken back. It's enough time that I sometimes double tap, which takes me all the way back to the Reader menu (also losing my place in the post list).

### Steps to reproduce the behavior

1. Open the WordPress app
2. Switch to the Reader tab
3. Select a post list (e.g. Followed Sites)
4. Open the fullpost view for a single post
5. Tap the back button in the top left to return to the post list -> Notice the slight delay
6. Tap a second time (while the fullpost view is still showing) -> You are taken back two screens to the Reader menu

**Comment by aerych** [comment]
Did some digging into this (props to @joelwills for the demo and additional testing) and it looks like this is situational, and more pronounced on older hardware like the iPhone 5. 

On a fresh feed, where only a single batch of posts have been cached, the delay is negligible if noticeable at all and the double tap behavior is difficult to reproduce.  However, on a feed where many batches of posts have been cached the issue is readily reproduced on iPhone 5 hardware. 

The delay appears to be due to the stream vc refreshing the list when the user taps the back button on the detail.  The list refresh takes a bit of time when the user might tap the back button again thinking the first tap did not register.  We'll need to review the need to refresh the full list (potentially unnecessary except for when a site was blocked).

**Comment by koke** [comment]
I've noticed this a lot lately on an iPhone 6, using the internal beta, and mostly on the a8c reader stream. The delay on the back button is annoying, but as the freeze is happening on the reader list's `viewWillAppear` this also blocks the drag-from-the-left-border gesture to go back, making the app look unresponsive. I'd often see delays of 4-5 seconds between when I started dragging and the UI reacted.
